fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1259,5 +1277,84 @@ public static String buildSearchSelfUri(String requestUriString, FHIRSearchConte
          */
         return UriBuilder.builder().context(context).requestUri(requestUriString).toSearchSelfUri();
     }
+    
+    /**
+     * Return a limited subset of elements from the resource. This subset SHOULD consist solely of all supported elements 
+     * that are marked as "summary" in the base definition of the resource(s)
+     * @param resourceType
+     * @return
+     */
+    public static Set<String> getSummaryElementNames(Class<?> resourceType) {
+        if (resourceType == null) {
+            return null;
+        }
+        if (resoureTypeSummaryElements.get(resourceType) != null) {
+            return Collections.unmodifiableSet(resoureTypeSummaryElements.get(resourceType));
+        } else {
+            StructureDefinition structureDefinition = FHIRRegistry.getInstance()
+                    .getResource("http://hl7.org/fhir/StructureDefinition/" + resourceType.getSimpleName(), StructureDefinition.class);
+            if (structureDefinition == null) {
+                return null;
+            } else {
+                Set<String> summaryElements = structureDefinition.getSnapshot().getElement().stream()
+                        .filter(elementDefinition -> elementDefinition.getIsSummary().getValue())
+                        .map(elementDefinition -> elementDefinition.getBase().getPath().getValue())
+                        .map(path -> path.substring(path.lastIndexOf(".") + 1).replace("[x]", ""))
+                        .collect(Collectors.toSet());
+                if (summaryElements != null) {
+                    resoureTypeSummaryElements.put(resourceType, summaryElements);
+                }
+                return Collections.unmodifiableSet(summaryElements);
+            }
+        }
+    }
+    
+    
+    
+    /**
+     * Return only the "text" element, the 'id' element, the 'meta' element, and only top-level mandatory elements
+     * The id, meta and the top-level mandatory elements will be added by the ElementFilter automatically.
+     * @param resourceType
+     * @return
+     */
+    public static Set<String> getSummaryTextElementNames(Class<?> resourceType) {
+        // Align with other getSummaryxxx functions, we may need the input resourceType in the future
+        Set<String> summaryTextList = new HashSet<String>();
+        summaryTextList.add("text");
+        return Collections.unmodifiableSet(summaryTextList);
+         
+    }
+    
+    
+    
+    /**
+     * Remove the text element
+     * @param resourceType
+     * @return
+     */
+    public static Set<String> getSummaryDataElementNames(Class<?> resourceType) {
```

Comment:
can we remove this method and use ModelSupport now?

Reply:
done!